src/benefits/benefits.service.ts

```diff
@@ -0,0 +1,57 @@
+import { BadRequestException, Injectable, InternalServerErrorException } from '@nestjs/common';
+import { HttpService } from '@nestjs/axios';
+import { SearchRequestDto } from './dto/search-request.dto';
+import { BENEFIT_CONSTANTS } from 'src/benefit.contants';
+import { ConfigService } from '@nestjs/config';
+
+@Injectable()
+export class BenefitsService {
+    private readonly strapiUrl: string;
+    private readonly strapiToken: string;
+    constructor(private readonly httpService: HttpService, private readonly configService: ConfigService) {
+        this.strapiUrl = this.configService.get<string>('STRAPI_URL') || '';
+        this.strapiToken = this.configService.get('STRAPI_TOKEN') || '';
+    }
+
+    onModuleInit() {
+        if (
+            !this.strapiToken.trim().length ||
+            !this.strapiUrl.trim().length
+        ) {
+            throw new InternalServerErrorException(
+                'Environment variables STRAPI_URL and STRAPI_TOKEN must be set',);
+        }
+    }
+
+    async searchBenefits(searchRequest: SearchRequestDto): Promise<any> {
+        if (searchRequest.context.domain === BENEFIT_CONSTANTS.FINANCE) {
+            // Example: Call an external API
+            const response = await this.httpService.axiosRef.get(`${this.strapiUrl}/benefits`,
```

Comment:
How do you plan to pass on search filters, pagination limits etc?

Reply:
I have added it in the dto when it will be required will handle accordingly